addon/components/kinetic-form.js

```diff
@@ -128,24 +137,34 @@ export default Component.extend({
     set(this, '_updatedFields', A());
   },
 
+  willDestroyElement() {
+    this._super(...arguments);
+    let debounced = get(this, '_debounced');
+    if (debounced) cancel(debounced);
+  },
+
```

Comment:
You can avoid this by using the ol fashioned `isDestroyed` check in the denounced functions.

```js
validateAndNotifyUpdate() {
  if (this.isDestroyed) { return; }
  ...
},
notifyUpdate() {
  if (this.isDestroyed) { return; }
  ...
}
```

And then you don't need `_debounced` or `willDestroyElement` anymore.

Reference: https://github.com/emberjs/ember.js/issues/3780

Now, if only we had ember-concurrency.

Reply:
@sukima, @cmackenz I'm ok to lean on ember-concurrency in this addon...  no issues there.

Reply:
@cmackenz I think this one may be worth doing

Reply:
@utilityboy Using `this.isDestroyed` will work just fine in this case.

I think that the easy is to just do that but keep an eye out for adding ember-concurrency.

In other words, let's add e-c when we need it. This instance is probably not important enough to need e-c yet.